crates/emitter/src/bytecode_offset.rs

```diff
@@ -0,0 +1,61 @@
+/// For tracking bytecode offsets in jumps
+#[derive(Clone, Copy, PartialEq, Debug)]
+#[must_use]
+pub struct BytecodeOffset {
+    pub offset: usize,
+}
+
+impl BytecodeOffset {
+    fn new(offset: usize) -> Self {
+        Self { offset }
+    }
+
+    /// diff_from is useful for finding the offset between two bytecodes. This is usually
+    /// used for jumps and jump_targets.
+    ///
+    /// For a forward jump, end will be a larger number, and start will be smaller
+    /// the output will be a positive number.  for a backward jump, the reverse
+    /// will be true, and the number will be negative. So it is important to use this api
+    /// consistently in both cases.
+    ///
+    /// Examples:
+    /// let offset_diff: BytecodeOffsetDiff = forward_jump_target_offset.diff_from(jump)
+    /// let offset_diff: BytecodeOffsetDiff = backward_jump_target_offset.diff_from(jump)
+    pub fn diff_from(self, start: BytecodeOffset) -> BytecodeOffsetDiff {
+        BytecodeOffsetDiff::new(self, start)
+    }
+}
+
+impl From<BytecodeOffset> for usize {
+    fn from(offset: BytecodeOffset) -> usize {
+        offset.offset
+    }
+}
+
+impl From<usize> for BytecodeOffset {
+    fn from(offset: usize) -> BytecodeOffset {
+        BytecodeOffset::new(offset)
+    }
+}
+
+pub struct BytecodeOffsetDiff {
+    diff: i32,
+}
+
+impl BytecodeOffsetDiff {
+    fn new(end: BytecodeOffset, start: BytecodeOffset) -> Self {
+        let diff = (end.offset as f64 - start.offset as f64) as i32;
```

Comment:
I think it's better using `i128` instead of `f64`.
`f64` cannot represent whole `usize` range.